### PR TITLE
Configuration of connection settings as URL

### PR DESCRIPTION
**Pull request by KingCrunch**
It would be nice, if I could configure the connection settings as a single string for example as an URL instead of 4 separate keys.

    //user:password@host:post/vhost

### CODE REVIEW

**Comment by KingCrunch** [comment]
Hi @videlalvaro 

I've converted this one into a PR (I don't think I can do this with #113, can I?). Feedback appreciated. Especially one cannot set the `ssl_context`-parameters via URL. This solution is inspired by the way the doctrine-bundle handles the URL, so like there one can omit every part (here: except the scheme and the host) and in that case the normal configuration options will be used again.

File: README.md [hunk 1]
```diff
@@ -120,6 +120,11 @@ old_sound_rabbit_mq:
 
             # requires php-amqplib v2.4.1+
             heartbeat: 0
+        another:
+            # A different (unused) connection defined by an URL. One can omit all parts,
+            # except the scheme (amqp:). If both segment in the URL and a key value (see above)
+            # are given the value from the URL takes precedence.
```
Comment (videlalvaro) [inline README.md]: why does the URL overrides what ones set above in a connection specifically? I would do it the other way around, the URL fills in values that might have not been specified in the connections confs

**Comment by videlalvaro** [comment]
Are you following this spec: https://www.rabbitmq.com/uri-spec.html ?

**Comment by KingCrunch** [comment]
Yes, at least I tried. Tell me, when I missed something. On the other hand the spec cover pretty common URIs, so there should be no bigger surprise.

**Comment by videlalvaro** [comment]
Any replies to my comment:

>why does the URL overrides what ones set above in a connection specifically? I would do it the other way around, the URL fills in values that might have not been specified in the connections confs

**Comment by videlalvaro** [comment]
@KingCrunch did you see my comments above? I would really like to merge your PR

**Comment by KingCrunch** [comment]
@videlalvaro Sorry, I indeed missed that comment.

The reason the URL overwrites the other settings is, that this is the way the `DoctrineBundle` works. I don't know, what's the idea, but I wanted to keep it in sync. Both are bundles and having two connections URLs, but both behaves differently would be confusing. However, that's nothing I'd fight for.

**Comment by videlalvaro** [comment]
@KingCrunch OK, if that's the way `DoctrineBundle` works, then we follow what the community is already doing. 

I'll review this again then.

**Comment by kesar** [comment]
👍 it would be great to have this merged

**Comment by dragosprotung** [comment]
Is there a reason this is not merged ?

**Comment by KingCrunch** [comment]
At least there were some conflicts now. I've rebased the branch, so somebody can review again and merge afterwards.

**Pull request merged**